### PR TITLE
Furnx 1064

### PR DESCRIPTION
**Pull request by jojicnikola**

### CODE REVIEW

**Review by jojicnikola — COMMENTED** [review]
@baguenth Demo is done

**Review by marijastosic — APPROVED** [review]

File: pos_operating_unit_sequence/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# Copyright 2024 Ametras
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html)
+
+{
+    "name": "PoS Order Sequence by Operating Unit",
+    "version": "13.0.1.0.0",
+    "license": "AGPL-3",
+    "category": "PoS",
+    "author": "Ametras intelligence GmbH, Odoo Community Association (OCA)",
+    "website": "https://www.ametras.com",
```
Comment (baguenth) [inline pos_operating_unit_sequence/__manifest__.py]: "website": "https://github.com/OCA/operating-unit",

File: pos_operating_unit_sequence/models/operating_unit.py [hunk 2]
```diff
@@ -0,0 +1,11 @@
+from odoo import fields, models
+
+
+class OperatingUnit(models.Model):
+    _inherit = "operating.unit"
+
+    pos_order_sequence_id = fields.Many2one(
+        comodel_name="ir.sequence",
+        string="PoS Order Config Sequence",
```
Comment (baguenth) [inline pos_operating_unit_sequence/models/operating_unit.py]: PoS Order Sequence

File: pos_operating_unit_sequence/models/operating_unit.py [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+        help="Sequence of pos order config with this operating unit",
```
Comment (baguenth) [inline pos_operating_unit_sequence/models/operating_unit.py]: Sequence of pos order with this operating unit

**Comment by baguenth** [comment]
@jojicnikola Please add license text in all files. Check other OCA modules as reference, how the specify license headers

**Review by baguenth — CHANGES_REQUESTED** [review]
@jojicnikola Please add license text in all files. Check other OCA modules as reference, how the specify license headers

File: purchase_operating_unit_sequence/__manifest__.py [hunk 4]
```diff
@@ -0,0 +1,17 @@
+# Copyright 2024 Ametras
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html)
+
+{
+    "name": "Purchase Order Sequence by Operating Unit",
+    "version": "13.0.1.0.0",
+    "license": "AGPL-3",
+    "category": "Sales",
+    "author": "Ametras intelligence GmbH, Odoo Community Association (OCA)",
+    "website": "https://www.ametras.com",
```
Comment (baguenth) [inline purchase_operating_unit_sequence/__manifest__.py]: "website": "https://github.com/OCA/operating-unit",

File: purchase_operating_unit_sequence/models/purchase_order.py [hunk 5]
```diff
@@ -0,0 +1,13 @@
+from odoo import api, models
+
+
+class PurchaseOrder(models.Model):
+    _inherit = "purchase.order"
+
+    @api.model
+    def create(self, vals):
+        if vals.get("operating_unit_id", False):
```
Comment (baguenth) [inline purchase_operating_unit_sequence/models/purchase_order.py:12]: I guess we should use
`if vals.get("name", "/") == "/" and vals.get("operating_unit_id", False):`

File: pos_operating_unit_sequence/models/pos_order.py [hunk 6]
```diff
@@ -0,0 +1,13 @@
+from odoo import models
+
+
+class PosOrder(models.Model):
+    _inherit = "pos.order"
+
+    def write(self, vals):
```
Comment (baguenth) [inline pos_operating_unit_sequence/models/pos_order.py]: In the original function we have the following:

`@api.model
  def create(self, values):
  session = self.env['pos.session'].browse(values['session_id'])
  values = self._complete_values_from_session(session, values)
  return super(PosOrder, self).create(values)`

`  @api.model
    def _complete_values_from_session(self, session, values):
    if values.get('state') and values['state'] == 'paid':
        values['name'] = session.config_id.sequence_id._next()
    values.setdefault('pricelist_id', session.config_id.pricelist_id.id)
    values.setdefault('fiscal_position_id', session.config_id.default_fiscal_position_id.id)
    values.setdefault('company_id', session.config_id.company_id.id)
    return values
`
  
Can we not simply inherit _complete_values_from_session to manipulate the name? 

In general i am wondering, if we need this module at all. 
Each pos-session clearly has a pos config in which we can define a sequence.

**Review by baguenth — CHANGES_REQUESTED** [review]

File: purchase_operating_unit_sequence/readme/CONTRIBUTORS.rst [hunk 7]
```diff
@@ -0,0 +1,6 @@
+
```
Comment (baguenth) [inline purchase_operating_unit_sequence/readme/CONTRIBUTORS.rst:2]: Well, that's not us...

**Review by baguenth — CHANGES_REQUESTED** [review]
@jojicnikola Please add license text in all files. Check other OCA modules as reference, how the specify license headers

File: purchase_operating_unit_sequence/models/purchase_order.py (hunk 5 above)
Comment (jojicnikola) [reply purchase_operating_unit_sequence/models/purchase_order.py:12]: In purchase.order we don't have vals.get("name") it's not set at all like in sale.order for example

**Review by baguenth — APPROVED** [review]

**Pull request merged**